### PR TITLE
feat(theme): theme prop is now available on each component

### LINKED ISSUE #13288: [Bug Report] Can't update theme in v-app v3-alpha
### Environment
**Vuetify Version:** 3.0.0-alpha.2
**Vue Version:** 3.0.5
**Browsers:** Chrome 89.0.4389.90
**OS:** Windows 10

### Steps to reproduce
Click on the "Change Theme" button to update the v-app and v-theme-provider theme.

You can change the defaultTheme to "light" to see what the light theme should look like. 

### Expected Behavior
The theme should update from dark to light when clicking on the button.

### Actual Behavior
The theme is not changing.

### Reproduction Link
<a href="https://codepen.io/Rpgilligan85/pen/MWbdLbj" target="_blank">https://codepen.io/Rpgilligan85/pen/MWbdLbj</a>

### Other comments
v-theme-provider works as expected but v-app is not updating the theme

I noticed that v-app used this, const { themeClasses } = useTheme()
and the v-theme-provider uses, const { themeClasses } = provideTheme(props, context)

### PR DESCRIPTION
**Pull request by nekosaur**
## Description

add `theme` prop to each component so that it can override provided value and start new context for it and all children

**NOTE** this will add two computed variables for every component. there is no way of conditionally providing values

fixes #13288 

## TODO
- Make sure documentation is up-to-date and correct

## How Has This Been Tested?

playground

## Markup:

<details>

```vue
<template>
  <v-app class="fill-height">
    <v-card :theme="theme" title="foo" subtitle="bar">
      <v-btn @click="theme = theme === 'light' ? 'dark' : 'light'">toggle</v-btn>
    </v-card>

    <v-card theme="dark" title="foo" subtitle="foo" color="purple-darken-2" append-icon="$close" />

    <v-theme-provider theme="dark" with-background>
      <v-card title="foo" subtitle="bar" class="ma-10" />
    </v-theme-provider>

    <v-card theme="dark" title="first">
      <v-card theme="light" title="second" class="ma-5">
        <v-card theme="dark" title="third" class="ma-5" />
      </v-card>
    </v-card>

    <v-card theme="dark" title="foo" subtitle="bar">
      <v-theme-provider theme="light">
        <v-card-actions>
          <v-btn color="primary">foo</v-btn>
        </v-card-actions>
      </v-theme-provider>
    </v-card>
  </v-app>
</template>

<script>
  import { ref } from 'vue'

  export default {
    name: 'Playground',
    setup () {
      const theme = ref('light')

      return { theme }
    },
  }
</script>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Review by johnleider — COMMENTED** [review]
Code looks good. Feedback is here https://discord.com/channels/340160225338195969/519331123239583755/847520162365833216

**Comment by glen-84** [comment]
Closes #13288. (?)

File: packages/docs/src/pages/en/features/theme.md [hunk 1]
```diff
@@ -95,7 +95,20 @@ export default {
 </script>
 ```
 
-Use the `<v-theme-provider>` component to dynamically apply different themes to *specific* sections of your application. In the following example, we apply a custom theme named `high-contrast`:
+Most components support the **theme** prop. When used, a new context will be created for the specific component and all of its children. In the example below the button will be using the **dark** theme since it is placed inside the card.
```
Comment (johnleider) [inline packages/docs/src/pages/en/features/theme.md:98]:
```suggestion
Most components support the **theme** prop. When used, a new context is created for _that_ specific component and **all** of its children. In the following example, the [v-btn](/components/buttons/) uses the **dark** theme applied by its parent [v-card](/components/cards/).
```

**Review by johnleider — CHANGES_REQUESTED** [review]
Looks good. Any api-generator updates that need to be made?

File: packages/docs/src/pages/en/components/theme-providers.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## API
+
+- [v-theme-provider](/api/v-theme-provider)
+
+## Examples
+
+### Background
+
+By default `v-theme-provider` will act as a renderless component, and simply provide a new theme context to all of its children. However if you use the **with-background** prop, it will render a `div` and apply the correct background color to it according to the theme used.
```
[the remark this answers is not in the document]
Comment (nekosaur) [reply packages/docs/src/pages/en/components/theme-providers.md:22]: 1. The link for renderless component goes to page on render functions, which does not contain any specific information about the concept of renderless components
2. Is it relevant for the user that theme-provider uses provide/inject?

[the remark this answers is not in the document]
Comment (johnleider) [reply packages/docs/src/pages/en/components/theme-providers.md:22]: 1. Thought they were the same. We can update the page link or remove it entirely if it doesn't make sense
2. It introduces the user to a concept we use throughout Vuetify 3
3. Don't understand but if it's just a language update, what should it change to?

[the remark this answers is not in the document]
Comment (nekosaur) [reply packages/docs/src/pages/en/components/theme-providers.md:22]: 1. I'd say remove. I don't think official vue docs talk about renderless anywhere
2. Not sure if this is the place to do it
3. `When you use the with-background prop the v-theme-provider will render a div and apply the background color or something something something`

[the remark this answers is not in the document]
Comment (johnleider) [reply packages/docs/src/pages/en/components/theme-providers.md:22]: 1. 👍 
2. Okay, I'll defer to your judgement
3. `When using the **with-background** prop, the `v-theme-provider` renders a `<div>` element and applies the specified theme's background color to the element.

File: packages/docs/src/pages/en/components/theme-providers.md [hunk 3]
```diff
@@ -19,6 +19,6 @@ The theme provider allows you to style a section of your application in a differ
 
 ### Background
 
-By default `v-theme-provider` will act as a renderless component, and simply provide a new theme context to all of its children. However if you use the **with-background** prop, it will render a `div` and apply the correct background color to it according to the theme used.
+By default, `v-theme-provider` is a renderless component that allows you to change the applied theme for all of its children. When using the **with-background** prop, the `v-theme-provider` will wrap its children in an element and apply the correct background color to it according to the theme used.
```
Comment (johnleider) [inline packages/docs/src/pages/en/components/theme-providers.md:22]:
```suggestion
By default, `v-theme-provider` is a renderless component that allows you to change the applied theme for all of its children. When using the **with-background** prop, the `v-theme-provider` wraps its children in a `<div>` element and applies selected **theme's** background color.
```

**Review by johnleider — CHANGES_REQUESTED** [review]

**Review by johnleider — APPROVED** [review]

**Pull request merged**